tests/system_tests_tcp_conns_terminate.py

```diff
@@ -0,0 +1,387 @@
+                              'port': echo_server_2.port,
+                              'address': cls.address_terminate,
+                              "closeConnectionsOnDelete": True}),
+            ('tcpListener', {'host': "0.0.0.0",
+                             'port': cls.listener_ports[router_2_id][cls.address_terminate],
+                             'address': cls.address_terminate,
+                             "closeConnectionsOnDelete": True}),
+        ])
+
+        cls.router_2 = cls.tester.qdrouterd('test_router_2', config_2, wait=True)
+        cls.router_1 = cls.tester.qdrouterd('test_router_1', config_1, wait=True)
+
+        cls.router_1.wait_router_connected('R2')
+        cls.router_2.wait_router_connected('R1')
+
+        cls.snooper_thread = VFlowSnooperThread(cls.router_1.addresses[0])
+
+        # wait for the TCP listeners and connectors
+        expected = {
+            router_1_id : [
+                ('LISTENER', {'VAN_ADDRESS': cls.address_terminate}),
+                ('LISTENER', {'VAN_ADDRESS': cls.address_default})
+            ],
+            router_2_id : [
+                ('LISTENER', {'VAN_ADDRESS': cls.address_terminate}),
+                ('CONNECTOR', {'VAN_ADDRESS': cls.address_terminate}),
+                ('LISTENER', {'VAN_ADDRESS': cls.address_default}),
+                ('CONNECTOR', {'VAN_ADDRESS': cls.address_default})
+            ]
+        }
+        success = retry(lambda: cls.snooper_thread.match_records(expected))
+        result = cls.snooper_thread.get_results()
+        cls.assertTrue(success, f"Failed to match records {result}")
+
+        # vflow ids are necessary to relate flows to tcp listeners and connectors
+        cls.router_1_vflow_id = None
+        cls.router_2_vflow_id = None
+        for router_key, router_attrs in result.items():
+            for rec in router_attrs:
+                if rec['RECORD_TYPE'] == 'ROUTER':
+                    if rec['NAME'].endswith(f"/{router_1_id}"):
+                        cls.router_1_vflow_id = rec['IDENTITY']
+                    elif rec['NAME'].endswith(f"/{router_2_id}"):
+                        cls.router_2_vflow_id = rec['IDENTITY']
+
+        # retry() parameters for VanFlowSnooper.match_record() tests where we
+        # expect failure. E.g. we try to match the 'END_TIME' attribute in the
+        # 'FLOW' record - and expect failure - in order to check that a flow
+        # is still acive.
+        cls.timeout = 1
+        cls.delay = 0.5
+
+    def get_tcp_entity_vflow_id(self, router_vflow_id, record_type, address):
+        res = self.snooper_thread.get_results()
+        try:
+            for rec in res[router_vflow_id]:
+                if rec['RECORD_TYPE'] == record_type and rec['PROTOCOL'] == 'tcp' and rec['VAN_ADDRESS'] == address:
+                    return rec['IDENTITY']
+        except KeyError:
+            pass
+        return None
+
+    def run_skmanage(self, cmd, router):
```

Comment:
There is a class in `system_test.py` called `QdManager` that runs the `skmanage` command for you. You can use that class instead of directly running `skmanage`

Reply:
I rebased and rebased and removed the run_skmanage() method.